packages/plugin-commands-publishing/src/publish.ts

```diff
@@ -51,8 +52,12 @@ export async function handler (
     engineStrict?: boolean,
     recursive?: boolean,
     workspaceDir?: string,
+    skipGitChecks?: boolean,
   } & Pick<Config, 'allProjects' | 'selectedProjectsGraph'>,
 ) {
+  if (!opts.skipGitChecks) {
+    await gitChecks()
+  }
```

Comment:
if the package is not inside a git repository, then the checks should be also skipped